designs/sam_publish_app_cmd.rst

```diff
@@ -0,0 +1,296 @@
+  https://console.aws.amazon.com/serverlessrepo/home?region=<region>#/published-applications/<arn>
+
+Create new version of an existing SAR application
+  Modify the existing template, change SemanticVersion to 0.0.2, and run ``sam publish app -t ./packaged.yaml`` again.
+  SAM CLI prints application metadata updated message, values of updated metadata and link to the console details page.
+
+  >>> sam publish app -t ./packaged.yaml
+  Publish Succeeded
+  The following metadata of application <id> has been updated:
+  {
+    "Author": "user1",
+    "Description": "description",
+    "ReadmeUrl": "s3://test/README.md",
+    ...
+    "SemanticVersion": "0.0.2",
+    "SourceCodeUrl": "https://github.com/hello"
+  }
+  Click the link below to view your application in AWS console:
+  https://console.aws.amazon.com/serverlessrepo/home?region=<region>#/published-applications/<arn>
+
+Update the metadata of an existing application without creating new version
+  Keep SemanticVersion unchanged, then modify metadata fields like Description or ReadmeUrl, and run
+  ``sam publish app -t ./packaged.yaml``. SAM CLI prints application metadata updated message, values of updated
+  metadata and link to the console details page.
+
+  >>> sam publish app -t ./packaged.yaml
+  Publish Succeeded
+  The following metadata of application <id> has been updated:
+  {
+    "Author": "qwang",
+    "Description": "description",
+    "ReadmeUrl": "s3://test/README.md"
+    ...
+  }
+  Click the link below to view your application in AWS console:
+  https://console.aws.amazon.com/serverlessrepo/home?region=<region>#/published-applications/<arn>
+
+Once the application is published, other developers in your team or your organization will be able to deploy it with a few
+clicks. If the application is shared publicly, the whole community will be able to find it by visiting the AWS Serverless
+Application Repository `public site`_.
+
+.. _public site: https://serverlessrepo.aws.amazon.com/applications
+
+
+Implementation
+==============
+
+CLI Changes
+-----------
+*Explain the changes to command line interface, including adding new commands, modifying arguments etc*
+
+1. Add a new top-level command called ``sam publish app`` with the following help message.
+
+.. code-block:: text
+
+  Usage: sam publish app [OPTIONS]
+
+    Use this command to publish a packaged AWS SAM template to the AWS
+    Serverless Application Repository to share within your team, across your
+    organization, or with the community at large.
+
+    This command expects the template's Metadata section to contain an
+    AWS::ServerlessRepo::Application section with application metadata
+    for publishing. For more details on this metadata section, see
+    https://docs.aws.amazon.com/serverlessrepo/latest/devguide/serverless-app-publishing-applications.html
+
+    Examples
+    --------
+    To publish an application
+    $ sam publish app -t packaged.yaml --region <region>
+
+  Options:
```

Comment:
Could we add the ability to share account IDs or chose to make the app completely public?